addons/sale/tests/test_sale_order_discount.py

```diff
@@ -108,6 +108,41 @@ def test_percent_discount_above_100(self):
         with self.assertRaises(ValidationError):
             self.wizard.write({'discount_percentage': 1.1, 'discount_type': 'sol_discount'})
 
+    def test_discount_translation(self):
+        self.env['res.lang']._activate_lang('es_AR')
+        self.wizard.write({
+            'discount_percentage': 0.1,
+            'discount_type': 'so_discount',
+        })
+        self.wizard.sale_order_id.partner_id.lang = 'es_AR'
+        discount_product = self.wizard._get_discount_product()
+        self.wizard.action_apply_discount()
+        discount_line = self.sale_order.order_line[-1]
+        self.assertEqual(discount_line.name, "Descuento: 10.00%")
+
+    def test_discount_translation_tax_groups(self):
+        self.env['res.lang']._activate_lang('es_AR')
+        self.wizard.write({
+            'discount_percentage': 0.1,
+            'discount_type': 'so_discount',
+        })
+        self.wizard.sale_order_id.partner_id.lang = 'es_AR'
+        tax1 = self.env['account.tax'].create({
+            'name': 'test_translation1',
+            'amount_type': 'percent',
+            'amount': 10})
+        tax2 = self.env['account.tax'].create({
+            'name': 'test_translation2',
+            'amount_type': 'percent',
+            'amount': 20})
```

Comment:
```suggestion
        tax1, tax2 = self.env['account.tax'].create([{
            'name': f"{percentage}% VAT",
            'amount_type': 'percent',
            'amount': percentage,
        }] for percentage in (10, 20))
```
Name strings need double quotes, and batching record creation is more efficient.